src/utils/create-slots.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react'
+import {useForceUpdate} from './use-force-update'
+
+/** createSlots is a factory that can create a
+ *  typesafe Slots + Slot pair to use in a component definition
+ *  For example: ActionList.Item uses createSlots to get a Slots wrapper
+ *  + Slot component that is used by LeadingVisual, Description
+ */
+const createSlots = <SlotNames extends string>(slotNames: SlotNames[]) => {
```

Comment:
This could be a useful standalone open source library eventually 💡

Reply:
One of these days 😉